### DISCUSSION TITLE
能否支持azure openai 的api

### DISCUSSION THREAD

**Discussion by TeamFan** [swuecho/chat#249] [Ideas]
或者能否提供base url选项😂谢谢作者

**Comment by swuecho** [comment]
已经有了 https://github.com/swuecho/chat/discussions/221

**Comment by swuecho** [comment]
网上搜了一下 azure api的格式, 适配了一下代码, 之前设计不太合理, 没有完全兼容

https://github.com/swuecho/chat/pull/250

你试一下不行我再看.

**Reply by TeamFan** [reply]
好像不太行，我用[这个项目](https://github.com/haibbo/cf-openai-azure-proxy) 转换了一下请求，虽然api地址和azure的api，好像是没有对应模型或者status500

**Comment by swuecho** [comment]
```
func DefaultAzureConfig(apiKey, baseURL, engine string) ClientConfig {
	return ClientConfig{
		authToken:  apiKey,
		BaseURL:    baseURL,
		OrgID:      "",
		APIType:    APITypeAzure,
		APIVersion: "2023-03-15-preview",
		Engine:     engine,

		HTTPClient: &http.Client{},

		EmptyMessagesLimit: defaultEmptyMessagesLimit,
	}
```
}

**Comment by swuecho** [comment]
这里面的值 你能提供几个?

我代码适配一下应该就可以了

**Reply by TeamFan** [reply]
要不我私发给你？

**Reply by swuecho** [reply]
参考这里, 感觉应该可以了 https://github.com/swuecho/chat/pull/261

如果不行 我试试弄个 azure账号

**Comment by zhaijunxiao** [comment]
感谢大佬，测试可以正常使用。
AZURE_RESOURCE_NAME 填这里的这个名字

https://oai.azure.com/deployment
![image](https://github.com/swuecho/chat/assets/40786117/bd663307-b857-4ecc-960e-ca5e6f8f6a31)

**Comment by swuecho** [comment]
@TeamFan 你可以参照上面的截图配置一下

**Reply by TeamFan** [reply]
谢谢大佬，我试一试，因为我railway突然给我封了，所以一直没有尝试，抱歉

**Reply by swuecho** [reply]
我的flyio 国内也无法访问了  太难了

**Reply by swuecho** [reply]
你可以docker 在国内用 azure 国内时可以用的

**Reply by TeamFan** [reply]
我试一试azure,谢谢大佬
